lib/optimizely/notification_center.rb

```diff
@@ -0,0 +1,154 @@
+        @logger.log Logger::ERROR, "#{notification_callback} is invalid."
+        return nil
+      end
+      
+      @notifications[notification_type].each do |notification|
+        return -1 if notification[:callback] == notification_callback
+      end
+      @notifications[notification_type].push(notification_id: @notification_id, callback: notification_callback)
+      notification_id = @notification_id
+      @notification_id += 1
+      notification_id
+    end
+
+    def remove_notification_listener(notification_id)
+      # Removes previously added notification callback
+
+      # Args:
+      #  notification_id:
+      # Returns:
+      #  The function returns true if found and removed, false otherwise
+      unless notification_id
+        @logger.log Logger::ERROR, 'Notification ID can not be empty.'
+        return nil
+      end
+      @notifications.each do |key, _array|
+        @notifications[key].each do |notification|
+          if notification_id == notification[:notification_id]
+            @notifications[key].delete(notification_id: notification_id, callback: notification[:callback])
+            return true
+          end
+        end
+      end
+      false
+    end
+
+    def clear_notifications(notification_type)
+      # Removes notifications for a certain notification type
+      #
+      # Args:
+      #  notification_type: key to the list of notifications
+  
+      return nil unless notification_type_valid?(notification_type)
+      
+      @notifications[notification_type] = []
+      @logger.log Logger::INFO, "All callbacks for notification type #{notification_type} have been removed."
+    end
+    
+    def clean_all_notifications
+      # Remove all notifications
+      @notifications.keys.each { |key| @notifications[key] = [] }
+    end
+
+    def fire_notifications(notification_type, *args)
+      # Fires off the notification for the specific event.  Uses var args to pass in a
+      # arbitrary list of parameter according to which notification type was fired
+
+      # Args:
+      #  notification_type: Type of notification to fire
+      #  args: list of arguments to the callback
+  
+      return nil unless notification_type_valid?(notification_type)
+
+      @notifications[notification_type].each do |notification|
+        begin
+          notification_callback = notification[:callback]
+          notification_callback.call(*args)
+          @logger.log Logger::INFO, "Notification #{notification_type} sent successfully."
+        rescue => e
+          @logger.log(Logger::ERROR, "Problem calling notify callback. Error: #{e}")
+          return nil
+        end
+      end
+      
+    end
+    
+    private
+    
+    def notification_type_valid?(notification_type)
+      # Validates notification type
+      
+      # Args:
+      #  notification_type: Type of notification to fire
+      
+      # Returns true if notification_type is present or valid,  false otherwise
```

Comment:
present or valid : implies wrong meaning. Make it "Returns true if notification_type is valid"